app/models/manageiq/consumption/showback_charge.rb

```diff
@@ -18,7 +19,25 @@ def clean_cost
   end
 
   def stored_data_event
-    self.stored_data = self.showback_event.data
+    self.stored_data = {}
+    self.stored_data[Time.now.utc] = self.showback_event.data
```

Comment:
Wouldn't you need to be able to call this with the time you want?

In case the charge is done in the past to allow old charges to be done.